File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        HashMap<String, BiFunction<Long, Long, Long>> mp = new HashMap<>();
        mp.put("+", (a, b) -> a + b);
        mp.put("-", (a, b) -> a - b);
        mp.put("*", (a, b) -> a * b);
        mp.put("/", (a, b) -> a / b);

        Stack<Long> st = new Stack<>();
        for(String s : tokens) {
            if(!mp.containsKey(s)) {
               st.push(Long.parseLong(s));
            }
            else {
                Long a = st.pop();
                Long b = st.pop();
                st.push(mp.get(s).apply(b, a));
            }
        }

        return st.pop().intValue();
    }
}

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution(object):
    def evalRPN(self, tokens):
        mp = {
            "+": lambda a, b: a + b,
            "-": lambda a, b: a - b,
            "*": lambda a, b: a * b,
            "/": lambda a, b: int(float(a)/b) 
        }

        st = deque()
        for s in tokens:
            if s not in mp:
                st.append(int(s))
            else:
                a = st.pop()
                b = st.pop()
                st.append(mp[s](b, a))
        
        return st.pop()
        